rocketmq-connect/rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/connectorwrapper/Worker.java

```diff
@@ -256,41 +278,25 @@ public synchronized void startTasks(Map<String, List<ConnectKeyValue>> taskConfi
                 Converter recordConverter = (Converter) converterClazz.newInstance();
 
                 if (task instanceof SourceTask) {
-                    DefaultMQProducer producer = new DefaultMQProducer();
-                    producer.setNamesrvAddr(keyValue.getString(RuntimeConfigDefine.NAMESRV_ADDR));
-                    String rmqProducerGroup = keyValue.getString(RuntimeConfigDefine.RMQ_PRODUCER_GROUP);
-                    if (StringUtils.isEmpty(rmqProducerGroup)) {
-                        rmqProducerGroup = connectConfig.getRmqProducerGroup();
-                    }
-                    producer.setProducerGroup(rmqProducerGroup);
-                    int operationTimeout = keyValue.getInt(RuntimeConfigDefine.OPERATION_TIMEOUT);
-                    if (operationTimeout <= 0) {
-                        producer.setSendMsgTimeout(connectConfig.getOperationTimeout());
+                    if (this.producer.getDefaultMQProducerImpl().getServiceState() == ServiceState.CREATE_JUST) {
```

Comment:
It seems that use the ServiceState to keep there's only one producer isn't a good choice, how about adding checkServiceState() method in here? in this method, you can control the producer was only started once, and can also be used to launch multiple tasks in parallel in the future.